src/common/decorators/swagger.ts

```diff
@@ -89,3 +93,21 @@ export const ApiPropertyOneOf = ({ union, ...apiPropertyOptions }: ApiPropertyOn
 
   return applyDecorators(ApiProperty({ ...apiPropertyOptions, oneOf: oneOfItems }));
 };
+
+/**
+ * A helper that functions like `ApiCreatedResponse`, that also adds an `ApiAccepted` response in case `webhookUrl` is passed
+ *
+ * @param options - these will be passed to the `ApiCreatedResponse` decorator
+ */
+export function ApiCreatedOrSubscriptionResponse(
+  options: ApiResponseOptions
+): ReturnType<typeof applyDecorators> {
+  return applyDecorators(
+    ApiCreatedResponse(options),
+    ApiAcceptedResponse({
+      description:
+        'Returned if `webhookUrl` is passed in the body. A response will be returned after the transaction has been validated. The result will be posted to the `webhookUrl` given when the transaction is completed',
+      type: NotificationPayloadModel,
```

Comment:
What does ApiExtraModels do? It seems like it's something to expose in the docs that isn't otherwise referenced directly.

It looks like NotificationPayloadModel shows up, so I think we don't need to. I am not entirely clear why its needed though.

https://docs.nestjs.com/openapi/types-and-parameters#extra-models

> HINT You only need to use @ApiExtraModels() once for a specific model class.

If we only need to call it once, then it should probably be used outside of a decorator thats called for every endpoint.